modules/codegen/src/test/scala/tests/generators/dropwizard/server/DropwizardContentTypesTest.scala

```diff
@@ -0,0 +1,95 @@
+package tests.generators.dropwizard.server
+
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.twilio.guardrail.generators.Java.Dropwizard
+import com.twilio.guardrail.{ Context, Server, Servers }
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import scala.collection.JavaConverters._
+import support.SwaggerSpecRunner
+
+class DropwizardContentTypesTest extends AnyFreeSpec with Matchers with SwaggerSpecRunner {
+  private val swagger: String =
+    s"""
+       |openapi: 3.0.1
+       |paths:
+       |  /foo:
+       |    post:
+       |      operationId: foo
+       |      responses:
+       |        204: {}
+       |        404:
+       |          content:
+       |            application/json:
+       |              schema:
+       |                type: object
+       |  /foo-multiple:
+       |    post:
+       |      operationId: fooMultiple
+       |      responses:
+       |        204:
+       |          content:
+       |            application/json:
+       |              schema:
+       |                type: object
+       |        404:
+       |          content:
+       |            text/plain:
+       |              schema:
+       |                type: string
+       |""".stripMargin
+
+  "Produces annotation should still be added when success response has no body, but errors do" in {
+    val (
+      _,
+      _,
+      Servers(Server(_, _, _, genResource :: Nil) :: Nil, _)
+    ) = runSwaggerSpec(swagger)(Context.empty, Dropwizard)
+
+    genResource
+      .asClassOrInterfaceDeclaration()
+      .getMembers
+      .asScala
+      .collectFirst({
+        case method: MethodDeclaration if method.getNameAsString == "foo" => method
+      })
+      .get
```

Comment:
I think SwaggerSpecRunner brings in `OptionValues` from scalatest, so you can say `.value` and get a test failure instead of an exception.

Reply:
I'll actually take this in the PR I'm opening shortly.